### PR TITLE
Fix multisyncmodeselector

### PR DESCRIPTION
**Pull request by MarekM25**
- Fix switching to full sync mode in multisyncmodeselector
- Limit the loop in BlockchainProcessor during fast sync transition

### CODE REVIEW

File: src/Nethermind/Nethermind.Consensus/Processing/BlockchainProcessor.cs [hunk 1]
```diff
@@ -66,6 +66,7 @@ public class BlockchainProcessor : IBlockchainProcessor, IBlockProcessingQueue
         private DateTime _lastProcessedBlock;
 
         private int _currentRecoveryQueueSize;
+        private const int MaxBlocksDuringFastSyncTransition = 2048;
```
Comment (asdacap) [inline src/Nethermind/Nethermind.Consensus/Processing/BlockchainProcessor.cs:69]: Are there cases where there could legitimately be a case where it need to process a lot of block? Assuming 33kb per block (which is what I measured currently from DB, not sure if that is accurate), I'd err towards having a higher limit like 8000, which should take 0.25 GB, more or less, which is also slightly more than a day's worth of blocks.

Comment (MarekM25) [reply src/Nethermind/Nethermind.Consensus/Processing/BlockchainProcessor.cs:69]: okay, make sense. I will increase it :)

File: src/Nethermind/Nethermind.Runner/NLog.config [hunk 2]
```diff
@@ -100,9 +100,9 @@
     <!--  <logger name="Consensus.Processing.BlockchainProcessor" final="true"/> -->
 
     <!-- if sync get stuck this is the best thing to enable the Trace on -->
-    <!-- <logger name="Synchronization.ParallelSync.MultiSyncModeSelector" minlevel="Trace" writeTo="file-async"/> -->
```
Comment (asdacap) [inline src/Nethermind/Nethermind.Runner/NLog.config:103]: Do you want to keep this?

File: src/Nethermind/Nethermind.Synchronization.Test/ParallelSync/MultiSyncModeSelectorFastSyncTests.cs [hunk 3]
```diff
@@ -474,6 +474,16 @@ public void Does_not_move_back_to_state_sync_mistakenly_when_in_full_sync_becaus
                 .TheSyncModeShouldBe(GetExpectationsIfNeedToWaitForHeaders(SyncMode.Full | SyncMode.FastHeaders));
         }
 
+        [Test]
+        public void When_state_sync_does_not_finished_then_sync_mode_should_be_full()
```
Comment (asdacap) [inline src/Nethermind/Nethermind.Synchronization.Test/ParallelSync/MultiSyncModeSelectorFastSyncTests.cs:478]: Do you mean "then_sync_mode_should_not_be_full"?

File: src/Nethermind/Nethermind.Runner/NLog.config [hunk 4]
```diff
@@ -100,9 +100,9 @@
...
-    <!-- <logger name="Synchronization.ParallelSync.MultiSyncModeSelector" minlevel="Trace" writeTo="auto-colored-console-async"/> -->
-    <!-- <logger name="Synchronization.ParallelSync.MultiSyncModeSelector" final="true"/> -->
+    <logger name="Synchronization.ParallelSync.MultiSyncModeSelector" minlevel="Trace" writeTo="file-async"/>
```
Comment (asdacap) [inline src/Nethermind/Nethermind.Runner/NLog.config:103]: Do you want to keep this?

File: src/Nethermind/Nethermind.Runner/NLog.config (hunk 2 above)
Comment (MarekM25) [reply src/Nethermind/Nethermind.Runner/NLog.config:103]: of course, not :) removing...

File: src/Nethermind/Nethermind.Synchronization.Test/ParallelSync/MultiSyncModeSelectorFastSyncTests.cs (hunk 3 above)
Comment (MarekM25) [reply src/Nethermind/Nethermind.Synchronization.Test/ParallelSync/MultiSyncModeSelectorFastSyncTests.cs:478]: yes, thx

**Review by asdacap — APPROVED** [review]

**Comment by dceleda** [comment]
This PR contains multiple PRs. Is that right?

**Comment by MarekM25** [comment]
> This PR contains multiple PRs. Is that right?

@dceleda merged master, so only my PR now

**Review by marcindsobczak — APPROVED** [review]

**Pull request merged**